### PR TITLE
Add multiple faucets/library upgrade/Refine code

### PR DESCRIPTION
**Pull request by sfffaaa**

### CODE REVIEW

**Comment by niazhussain** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=peaqnetwork%2Fsubstrate-faucet).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/peaqnetwork/substrate-faucet/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by niazhussain** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=peaqnetwork%2Fsubstrate-faucet).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/peaqnetwork/substrate-faucet/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by niazhussain** [comment]
@sfffaaa: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr | 9e6aaff084c884b915c0b94b35c6ff81eda2c298 | [link](http://dashboard-jx.ci.peaq.network/namespaces/jx/pipelineruns/network-substrate-faucet-pr-2-pr-zdnw5) | `/test pr`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by niazhussain** [comment]
@sfffaaa: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr | 9e6aaff084c884b915c0b94b35c6ff81eda2c298 | [link](http://dashboard-jx.ci.peaq.network/namespaces/jx/pipelineruns/network-substrate-faucet-pr-2-pr-h9c9c) | `/test pr`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: app/faucet.py [hunk 1]
```diff
@@ -1,110 +1,141 @@
...
+                reply = ""
+                extrinsic = substrate.create_signed_extrinsic(call=call, keypair=faucet_kp)
+                reply = substrate.submit_extrinsic(extrinsic, wait_for_inclusion=False)
+                extrinsic_url.append(
+                    '{1} on {0}'.format(rpc_url, reply['extrinsic_hash']))
+
         # inform user
-        msg = " Awesome, you just received \"{}\" {}! The extrinsic hash is [{}]".format(TOKENS_TO_SEND/1000000000000000000, TOKEN_SYMBOL, reply['extrinsic_hash'])
+        msg = " Awesome, you just received {} Token! Please check these extrinsic hash. \n {}".format(
+            TOKENS_TO_SEND / 1000000000000000000,
```
Comment (irediaes) [inline app/faucet.py]: The figure (1000000000000000000) should be added to env or configurable file. 
It's also duplicated,

File: app/faucet.py [hunk 2]
```diff
@@ -1,110 +1,141 @@
...
         await ctx.send(str(ctx.author.mention) + msg)
-        
+
         # log to console
         print(str(ctx.author.name) + msg)
-        
+
         # store metrics
-        ISSUANCE_TOTAL.inc(TOKENS_TO_SEND/1000000000000000000)
+        ISSUANCE_TOTAL.inc(TOKENS_TO_SEND / 1000000000000000000)
```
Comment (irediaes) [inline app/faucet.py]: The figure (1000000000000000000) should be added to env or configurable file. 
It's also duplicated,

**Review by irediaes — CHANGES_REQUESTED** [review]

**Comment by niazhussain** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=peaqnetwork%2Fsubstrate-faucet).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/peaqnetwork/substrate-faucet/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by niazhussain** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=peaqnetwork%2Fsubstrate-faucet).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/peaqnetwork/substrate-faucet/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by niazhussain** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=peaqnetwork%2Fsubstrate-faucet).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/peaqnetwork/substrate-faucet/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by niazhussain** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=peaqnetwork%2Fsubstrate-faucet).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/peaqnetwork/substrate-faucet/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by niazhussain** [comment]
@sfffaaa: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr | 2a0db1f7a75db5af60bdf1e1c8bda9d183baae9c | [link](http://dashboard-jx.ci.peaq.network/namespaces/jx/pipelineruns/network-substrate-faucet-pr-2-pr-vr9pq) | `/test pr`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by niazhussain** [comment]
@sfffaaa: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr | 2a0db1f7a75db5af60bdf1e1c8bda9d183baae9c | [link](http://dashboard-jx.ci.peaq.network/namespaces/jx/pipelineruns/network-substrate-faucet-pr-2-pr-4cvjd) | `/test pr`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by niazhussain** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=peaqnetwork%2Fsubstrate-faucet).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/peaqnetwork/substrate-faucet/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by niazhussain** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=peaqnetwork%2Fsubstrate-faucet).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/peaqnetwork/substrate-faucet/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by niazhussain** [comment]
@sfffaaa: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr | 24a19567044a7ecea232683964c4f74676a7e3a1 | [link](http://dashboard-jx.ci.peaq.network/namespaces/jx/pipelineruns/network-substrate-faucet-pr-2-pr-qfvkf) | `/test pr`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: app/faucet.py (hunk 1 above)
Comment (sfffaaa) [reply app/faucet.py]: Extract to env

File: app/faucet.py (hunk 2 above)
Comment (sfffaaa) [reply app/faucet.py]: Extract to env

**Review by irediaes — APPROVED** [review]

**Comment by niazhussain** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *[irediaes](https://github.com/peaqnetwork/substrate-faucet/pull/2#pullrequestreview-1305927672 "Approved")*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=peaqnetwork%2Fsubstrate-faucet).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/peaqnetwork/substrate-faucet/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by niazhussain** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *[irediaes](https://github.com/peaqnetwork/substrate-faucet/pull/2#pullrequestreview-1305927672 "Approved")*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=peaqnetwork%2Fsubstrate-faucet).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/peaqnetwork/substrate-faucet/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by niazhussain** [comment]
@sfffaaa: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr | 24a19567044a7ecea232683964c4f74676a7e3a1 | [link](http://dashboard-jx.ci.peaq.network/namespaces/jx/pipelineruns/network-substrate-faucet-pr-2-pr-bfnbr) | `/test pr`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Pull request merged**